econml/dml.py

```diff
@@ -134,6 +134,26 @@ def effect(self, T0, T1, X):
             T1 = self._one_hot_encoder.transform(reshape(self._label_encoder.transform(T1), (-1, 1)))[:, 1:]
         return super().effect(T0, T1, X)
 
+    def score(self, Y, T, X, W=None):
+        if T.ndim == 1:
+            T = reshape(T, (-1, 1))
+        if Y.ndim == 1:
+            Y = reshape(Y, (-1, 1))
+        if W is None:
+            W = np.empty((shape(Y)[0], 0))
+        Y_test_pred = np.zeros(shape(Y) + (self._n_splits,))
+        T_test_pred = np.zeros(shape(T) + (self._n_splits,))
+        for ind in range(self._n_splits):
+            Y_test_pred[:, :, ind] = reshape(self._models_y[ind].predict(X, W), shape(Y))
+            T_test_pred[:, :, ind] = reshape(self._models_t[ind].predict(X, W), shape(T))
+        Y_test_pred = Y_test_pred.mean(axis=2)
```

Comment:
Is the correct measurement to average the predictions first (as you do here) and then calculate the average MSE per row, or calculate the MSE for all first stage models and then average over them after?

If this calculation is the correct one, it may be worth putting in a comment explaining why this is better.

Reply:
@vasilismsr what do you think? Let me clarify the two options here:

1. first get Y test prediction and T test prediction by averaging the result from models_y and models_t, then get residuals and run final model to get MSE. (The way I did it here.)

2. for each Y test prediction and T test prediction from models_y[i] and models_t[i], get residuals and run final model to get MSE[i], averaging MSE to get final MSE.

Reply:
Hm. I see what Keith is proposing. It might actually be a better way. But hard to say. I suspect since when we fit the true model we never used the average of the nuisance predictions, but we either used the one or the other nuisance on each data point, then maybe evaluating each nuisance individually and then averaging the qualities is better than averaging the predictions and then deriving a single quality metric for the average prediction. Both seem ok to me at a high level. I don't see any theoretical justification towards one or the other. Both seem correct theoretically from a worst-case performance perspective.

Reply:
I tried both methods on simulation data, the method Kieth proposed always gave a larger MSE (sometimes double the original MSE), but the model sequence is consistent with the original method sorted by MSE. It makes sense since the MSE from single nuisance prediction should be larger than MSE after averaging. I didn't find any case they show different model selection result. It's hard to say which one is better. BTW, on the OJ data, these two give pretty similar MSE.